Extension/YapoSubModule.cs

```diff
@@ -42,18 +41,33 @@ protected override void OnGameStart(Game game, IGameStarter gameStarterObject)
             base.OnGameStart(game, gameStarterObject);
             if (!(game.GameType is Campaign)) return;
 
-            MBObjectManager.Instance.RegisterType<TroopSorterService>("Sorter", "Sorters");
-            
-            (gameStarterObject as CampaignGameStarter)?.LoadGameTexts($"{BasePath.Name}/Modules/{Strings.MODULE_FOLDER_NAME}/{Strings.MODULE_DATA_PARTY_COUNT_STRINGS}");
+            try
+            {
+                MBObjectManager.Instance.RegisterType<TroopSorterService>("Sorter", "Sorters");
+                AddBehaviours(gameStarterObject as CampaignGameStarter);
+            } catch (Exception exception)
+            {
+                ModDebug.ShowError("Something went wrong while starting a game", "OnGameStart exception", exception);
+            }
+        }
+
+        private static void AddBehaviours(CampaignGameStarter campaignGameStarter)
+        {
+            if (campaignGameStarter == null) return;
+
+            if (YapoSettings.Instance.IsFormationPersistenceFixEnabled)
+            {
+                campaignGameStarter.AddBehavior(FixedFormationsBehaviour.Instance);
+                campaignGameStarter.LoadGameTexts($"{BasePath.Name}/Modules/{Strings.MODULE_FOLDER_NAME}/{Strings.MODULE_DATA_FORMATION_STRINGS}");
+            }
+
+            campaignGameStarter.LoadGameTexts($"{BasePath.Name}/Modules/{Strings.MODULE_FOLDER_NAME}/{Strings.MODULE_DATA_PARTY_COUNT_STRINGS}");
```

Comment:
Looks to me like this is the only real change in this PR?

Reply:
Bulk of additions are in the two new files `Extension/Services/PartyHeaderCountHelper.cs` and `Extension/Patches/PartyVmPatches.cs`